config/dependency.go

```diff
@@ -776,11 +777,18 @@ func getTerragruntOutputJSON(ctx *ParsingContext, targetConfig string) ([]byte,
 		return nil, err
 	}
 
+	// Fetch engine options so they can be passed to the dependency functions
+	engineOpts, err := remoteStateTGConfig.EngineOptions()
```

Comment:
I think `engineOpts` can be persisted to `ctx.TerragruntOptions.Engine` to be used later from any place in the code